### PR TITLE
Feature: Separation of storage backend and base table

### PR DESCRIPTION
**Pull request by arazthexd**
## Summary by CodeRabbit

* **New Features**
  * Multi-axis indexing constants (META/M, OBJ/O, FEAT/F).
  * Pluggable storage backend API and an eager in-memory backend with view, save/load and persistence support.

* **Improvements**
  * Unified table API with backend delegation, stronger validation, and improved subset/copy/concat/history semantics.
  * Cleaner package exports and consistent internal module organization.

* **Tests**
  * Large expansion of tests covering backends, table operations, persistence, and edge cases.

### CODE REVIEW

File: src/druglab/db/backend/base.py [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+            Feature array shape.
+        """
+
+    def _n_feature_rows(self) -> int:
+        """
+        Return the number of rows in each feature array.
+        """
+        if not self.get_feature_names():
+            return len(self)
+        return self.get_feature_shape(self.get_feature_names()[0])[0]
```
[the remark this answers is not in the document]
Comment (arazthexd) [reply src/druglab/db/backend/base.py:408]: It depends on len(self) because the mixin will always be part of a storage backend class which has __len__ defined.
